plutus-tx/src/PlutusTx/Show.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module PlutusTx.Show (
+    Show (..),
+) where
+
+import PlutusTx.Base
+import PlutusTx.Bool
+import PlutusTx.Builtins
+import PlutusTx.Builtins.Internal qualified
+import PlutusTx.Foldable (foldl)
+import PlutusTx.Show.TH (Show (..), ShowS, showString)
+
+instance Show Integer where
+    {-# INLINEABLE show #-}
+    show = displayConstant
```

Comment:
Do we really want to mix `prettyprinter`-based stuff into otherwise perfectly regular Haskell `Show`?

Reply:
Good question but I think it's orthogonal to this PR. The question is what instance the (to-be-added) builtin function(s) for showing values should use - I think it's probably better to use `Show` instead of `PrettyConst` there.

Reply:
Although `Show Data` probably doesn't look very good (especially the ByteString part), but I guess we could use a manual `Show Data` instance.